src/lib/optimizer/join_ordering/join_graph_builder.cpp

```diff
@@ -83,15 +83,10 @@ void JoinGraphBuilder::_traverse(const std::shared_ptr<AbstractLQPNode>& node) {
        */
 
       const auto union_node = std::static_pointer_cast<UnionNode>(node);
+      const auto parse_result = _parse_union(union_node);
```

Comment:
Could you please explain the changes in this file?

Reply:
(Not in the code, but here)

Reply:
The original thought was that the distinction of UnionModes didn't make sense in the first place since there was only one. I basically just wanted to see if something bad happens if the UnionModes are treated the same there and then forgot about it.
Anyway, I don't really get yet why we should only parse and traverse over a UnionPositions, but not UnionAll.

Reply:
Is this covered by any test at all?

Reply:
Actually, the UnionAll doesn't appear in any test (except for UnionAllTest of course)

Reply:
Did I understand you right that you wanted to look into that? Otherwise, we should open an issue.